src/TaskQueue.js

```diff
@@ -0,0 +1,43 @@
+const iterate = require('taskcluster-lib-iterate');
+const assert = require('assert');
+class TaskQueue {
+  constructor(cfg, queue, type) {
+    assert(queue, 'Instance of taskcluster queue is required');
+    this.queue = queue;
+    this.workerType = type;
+    this.provisionerId = cfg.worker.provisionerId;
+    this.workerGroup = cfg.worker.workerGroup;
+    this.workerId = type;
+  }
+
+  async runWorker() {
+    while (true) {
+      await this.claimTask();
+    }
+  }
+
+  async claimTask() {
+    let result = await this.queue.claimWork(this.provisionerId, this.workerType, {
+      tasks: 1,
+      workerGroup: this.workerGroup,
+      workerId: this.workerId,
+    });
+    if (result.tasks.length === 0) {
+      return ;
+    }
+    const task = result.tasks[0];
```

Comment:
I'd maybe explain why are we interested in the first element of the array in the comment, but it shouldn't block the PR